client/app/pages/queries/add-to-dashboard.js

```diff
@@ -0,0 +1,104 @@
+import template from './add-to-dashboard.html';
+
+const AddToDashboardForm = {
+  controller($sce, Dashboard, currentUser, toastr, Query, Widget) {
+    'ngInject';
+
+    this.query = this.resolve.query;
+    this.saveAddToDashbosard = this.resolve.saveAddToDashboard;
+    this.saveInProgress = false;
+
+    this.trustAsHtml = html => $sce.trustAsHtml(html);
+
+    this.onDashboardSelected = (dash) => {
+      // console.log(dash);
+      // add widget to dashboard
+      this.saveInProgress = true;
+
+      // this.dashboard = this.resolve.dashboard;
+      // this.saveInProgress = false;
+      this.widgetSize = 1;
+      this.selectedVis = null;
+      this.query = {};
+      this.selected_query = this.query.id;
+
+      this.type = 'visualization';
+      this.isVisualization = () => this.type === 'visualization';
+
+      // TODO handle multiple visualization queries
+
+      // console.log(this.resolve.query);
+      Query.get({ id: this.resolve.query.id }, (query) => {
+        if (query) {
+          this.selected_query = query;
+          if (query.visualizations.length) {
+            this.selectedVis = query.visualizations[0];
+          }
+        }
+      });
+
+      const widget = new Widget({
+        visualization_id: this.selectedVis && this.selectedVis.id,
+        dashboard_id: dash.id,
+        options: {},
+        width: this.widgetSize,
+        type: this.type,
+      });
+      console.log(dash);
+      console.log(widget);
+      widget.$save().then((response) => {
+        console.log(response);
+        this.selectedDashboard = Dashboard.get({ slug: dash.slug });
```

Comment:
It seems Dashboard.get() returns a promise in which case you simply have to use it upon its resolved state. I see in one of Arik's cases something like (you can ignore the throttle). I didn't get to do a deeper dive into what 'get()' is but promises can pretty much only be used once so it explains the first working console log...even if I am wrong you should try this approach:

`this.loadDashboard = _.throttle((force) => {
    this.dashboard = Dashboard.get({ slug: $routeParams.dashboardSlug }, (dashboard) => {
      Events.record('view', 'dashboard', dashboard.id);
      renderDashboard(dashboard, force);
    }, () => {
        // error...
        // try again. we wrap loadDashboard with throttle so it doesn't happen too often.
        // we might want to consider exponential backoff and also move this as a general
        // solution in $http/$resource for all AJAX calls.
      this.loadDashboard();
    });
  }, 1000);`